CDTDatastore/touchdb/TDReplicator.m

```diff
@@ -819,9 +907,14 @@ - (void)saveLastSequence
                       // read-only and don't attempt to read its checkpoint next time.
                   } else if (_db) {
                       id rev = response[@"rev"];
+                      id ID = response[@"id"];
                       if (rev) body[@"_rev"] = rev;
+                      if (ID) body[@"_id"] = ID;
                       self.remoteCheckpoint = body;
-                      [_db setLastSequence:_lastSequence withCheckpointID:checkpointID];
+                      if (ID && [self.db saveCheckpointDocument:body error:nil]) {
+                          CDTLogWarn(CDTREPLICATION_LOG_CONTEXT,
```

Comment:
ok let's leave the behaviour consistent. I guess worst case we just end up replicating some stuff again - as long as we are at least confident that this isn't always failing "silently" in our tests say.